src/devices/mindy.ts

```diff
@@ -0,0 +1,366 @@
+import {Zcl} from "zigbee-herdsman";
+
+import * as exposes from "../lib/exposes";
+import * as m from "../lib/modernExtend";
+import {type DefinitionWithExtend, type Fz, KeyValueAny, type Tz} from "../lib/types";
+import * as utils from "../lib/utils";
+
+import * as globalStore from "../lib/store";
+
+const e = exposes.presets;
+const ea = exposes.access;
+
+const tzLocal = {
+    last_boot: {
+        key: ["last_boot"],
+        convertGet: async (entity, key, meta) => {
+            await entity.read("genTime", ["lastSetTime"]);
+        },
+    } satisfies Tz.Converter,
+    wifi: {
+        key: ["wifi"],
+        convertGet: async (entity, key, meta) => {
+            await entity.read("genBasic", ["locationDesc"]);
+        },
+        convertSet: async (entity, key, value, meta) => {
+            await entity.write("genBasic", {locationDesc: value === "ON" ? "1" : "0"});
+
+            const retryRead = async (attempts: number) => {
```

Comment:
Why is there a retry mechanism here?

Reply:
The device reboots after the wifi is turned on or off. Therefore, in order to get information about the changed IP address as quickly as possible (and the IP address or an empty string is transmitted through the location description).
in some networks one request is not enough, because in some cases we get an error in response.